### PR TITLE
To fix the error when trying to install the ibmcloud CLI

### PR DESCRIPTION
**Pull request by aishwaryabk**
Modification to #178

### CODE REVIEW

File: openshift-install-powervs [hunk 1]
```diff
@@ -1146,9 +1146,11 @@ function setup_terraform {
   else
     log "Installing the latest version of Terraform..."
     if [[ "${ARCH}" == "ppc64le" ]]; then
-      retry "curl --connect-timeout 30 -fsSL https://oplab9.parqtec.unicamp.br/pub/ppc64el/terraform/terraform-${TF_LATEST:1} -o ${TF}"
+      #retry "curl --connect-timeout 30 -fsSL https://oplab9.parqtec.unicamp.br/pub/ppc64el/terraform/terraform-${TF_LATEST:1} -o ${TF}"
+      retry "curl --connect-timeout 30 -fsSL https://oplab9.parqtec.unicamp.br/pub/ppc64el/terraform/terraform-0.13.6 -o ${TF}"
     else
-      retry "curl --connect-timeout 30 -fsSL https://releases.hashicorp.com/terraform/${TF_LATEST:1}/terraform_${TF_LATEST:1}_${OS}_amd64.zip -o ./terraform.zip"
+      #retry "curl --connect-timeout 30 -fsSL https://releases.hashicorp.com/terraform/${TF_LATEST:1}/terraform_${TF_LATEST:1}_${OS}_amd64.zip -o ./terraform.zip"
+      retry "curl --connect-timeout 30 -fsSL https://releases.hashicorp.com/terraform/0.13.6/terraform_0.13.6_linux_amd64.zip -o ./terraform.zip"
```
Comment (cs-zhang) [inline openshift-install-powervs:1153]: What's the reason to make such change to use hard coded version?

File: openshift-install-powervs [hunk 2]
```diff
@@ -1186,7 +1188,21 @@ function setup_poweriaas() {
...
+
+  ARCH=$(uname -m)
+  if [ "$ARCH" == "x86_64" ]; then
+  	ARCH="amd64"
+  fi
+
+  OS_ARCH="${OS_NAME}_${ARCH}"
+
+  CLI_LATEST=$(echo ${TAGS[${#TAGS[@]}-1]} | awk -F "/" '{print $3}' | tr -d 'v')
+
```
Comment (cs-zhang) [inline openshift-install-powervs:1205]: The old way to get CLI_LATEST works fine, why do we need to use the new way to get CLI_LATEST value?

File: openshift-install-powervs (hunk 1 above)
Comment (aishwaryabk) [reply openshift-install-powervs]: Updated the changes

File: openshift-install-powervs [hunk 3]
```diff
@@ -1148,7 +1148,7 @@ function setup_terraform {
     if [[ "${ARCH}" == "ppc64le" ]]; then
       retry "curl --connect-timeout 30 -fsSL https://oplab9.parqtec.unicamp.br/pub/ppc64el/terraform/terraform-${TF_LATEST:1} -o ${TF}"
     else
-      retry "curl --connect-timeout 30 -fsSL https://releases.hashicorp.com/terraform/${TF_LATEST:1}/terraform_${TF_LATEST:1}_${OS}_amd64.zip -o ./terraform.zip"
+      retry "curl --connect-timeout 30 -fsSL https://releases.hashicorp.com/terraform/${TF_LATEST:1}/terraform_${TF_LATEST:1}_${OS,}_amd64.zip -o ./terraform.zip"
```
Comment (cs-zhang) [inline openshift-install-powervs:1151]: Is the ${OS,}_amd64.zip correct?

File: openshift-install-powervs [hunk 4]
```diff
@@ -1186,7 +1186,21 @@ function setup_poweriaas() {
 # If latest is available in System PATH then use symlink
 #-------------------------------------------------------------------------
 function setup_ibmcloudcli() {
-  CLI_LATEST=$(curl -s https://api.github.com/repos/IBM-Cloud/ibm-cloud-cli-release/releases/latest | grep tag_name | cut -d'"' -f4 | sed 's/v//')
+  TAGS=( $(git -c 'versionsort.suffix=-' ls-remote --exit-code --refs --sort='version:refname' --tags \
+  https://github.com/IBM-Cloud/ibm-cloud-cli-release '*.*.*') )
+
+  OS=$(uname -s);
```
Comment (yussufsh) [inline openshift-install-powervs:1192]: Why uname? Did you check what it returns on other operating systems eg: windows?

File: openshift-install-powervs (hunk 3 above)
Comment (yussufsh) [reply openshift-install-powervs:1151]: @aishwaryabk can you explain why we need a comma (,)?

File: openshift-install-powervs [hunk 5]
```diff
@@ -639,7 +639,7 @@ function powervs_login {
   log "Trying to login with the provided IBMCLOUD_API_KEY..."
   debug_switch
   $CLI_PATH login --apikey "$IBMCLOUD_API_KEY" -q --no-region > /dev/null
-  debug_switch
+  #debug_switch
```
Comment (yussufsh) [inline openshift-install-powervs:642]: This is to toggle back the debug switch at line 640. Why are we commenting it?

File: openshift-install-powervs (hunk 4 above)
Comment (yussufsh) [reply openshift-install-powervs:1192]: Also OS is global variable you don't want to override it here.

File: openshift-install-powervs (hunk 2 above)
Comment (yussufsh) [reply openshift-install-powervs:1203]: Same confusion here.

**Review by yussufsh — CHANGES_REQUESTED** [review]

**Comment by aishwaryabk** [comment]
Updated the changes.

**Comment by yussufsh** [comment]
closing with reference to #180

**Pull request closed without merging**